tests/unit/preprocessor/_multimodel/test_multimodel.py

```diff
@@ -1,267 +1,314 @@
-
-    def test_compute_overlap_statistic_yr_cube(self):
-        data = [self.cube4, self.cube4]
-        stats = multi_model_statistics(products=data,
-                                       statistics=['mean'],
-                                       span='overlap')
-        expected_ovlap_mean = np.ma.ones((2, 3, 2, 2))
-        self.assert_array_equal(stats['mean'].data, expected_ovlap_mean)
-
-    def test_compute_std(self):
-        """Test statistic."""
-        data = [self.cube1.data[0], self.cube2.data[0] * 2]
-        stat = _compute_statistic(data, "std")
-        expected = np.ma.ones((3, 2, 2)) * 0.5
-        expected[0, 0, 0] = 0
-        self.assert_array_equal(stat, expected)
-
-    def test_compute_max(self):
-        """Test statistic."""
-        data = [self.cube1.data[0] * 0.5, self.cube2.data[0] * 2]
-        stat = _compute_statistic(data, "max")
-        expected = np.ma.ones((3, 2, 2)) * 2
-        expected[0, 0, 0] = 0.5
-        self.assert_array_equal(stat, expected)
-
-    def test_compute_min(self):
-        """Test statistic."""
-        data = [self.cube1.data[0] * 0.5, self.cube2.data[0] * 2]
-        stat = _compute_statistic(data, "min")
-        expected = np.ma.ones((3, 2, 2)) * 0.5
-        self.assert_array_equal(stat, expected)
-
-    def test_compute_percentile(self):
-        """Test statistic."""
-        data = [self.cube1.data[0] * 0.5, self.cube2.data[0] * 2]
-        stat = _compute_statistic(data, "p75")
-        expected = np.ma.ones((3, 2, 2)) * 1.625
-        expected[0, 0, 0] = 0.5
-        self.assert_array_equal(stat, expected)
-
-    def test_put_in_cube(self):
-        """Test put in cube."""
-        cube_data = np.ma.ones((2, 3, 2, 2))
-        stat_cube = _put_in_cube(self.cube1, cube_data, "mean", t_axis=[1, 2])
-        self.assert_array_equal(stat_cube.data, self.cube1.data)
-
-    def test_assemble_overlap_data(self):
-        """Test overlap data."""
-        comp_ovlap_mean = _assemble_data([self.cube1, self.cube1],
-                                         "mean",
-                                         span='overlap')
-        expected_ovlap_mean = np.ma.ones((2, 3, 2, 2))
-        self.assert_array_equal(comp_ovlap_mean.data, expected_ovlap_mean)
-
-    def test_assemble_full_data(self):
-        """Test full data."""
-        comp_full_mean = _assemble_data([self.cube1, self.cube2],
-                                        "mean",
-                                        span='full')
-        expected_full_mean = np.ma.ones((5, 3, 2, 2))
-        expected_full_mean.mask = np.ones((5, 3, 2, 2))
-        expected_full_mean.mask[1] = False
-        self.assert_array_equal(comp_full_mean.data, expected_full_mean)
-
-    def test_plev_fix(self):
-        """Test plev fix."""
-        fixed_data = _plev_fix(self.cube2.data, 1)
-        expected_data = np.ma.ones((3, 2, 2))
-        self.assert_array_equal(expected_data, fixed_data)
-
-    def test_unify_time_coordinates(self):
```

Comment:
I think we might want to keep (or re-implemen) this test. Eventually I think this will be moved to the time preprocessor tests in https://github.com/ESMValGroup/ESMValCore/issues/744

Reply:
I thought this was already covered elsewhere. I re-implemented it and added it back. 👍